code/modules/clothing/masks/transmog.dm

```diff
@@ -0,0 +1,115 @@
+/obj/item/clothing/mask/morphing
+	name = "curious mask"
+	desc = "It doesn't really resemble anything, though it gives you an eerie and intense feeling nonetheless."
+	icon_state = "morphing_mask"
+	item_state = "chapmask"
+	flags = FPRINT
+	body_parts_covered = FACE
+	w_class = W_CLASS_SMALL
+	var/target_type = null
+
+/obj/item/clothing/mask/morphing/equipped(M as mob, wear_mask)
+	if(target_type)
+		var/mob/living/carbon/C = M
+		if(!istype(C))
+			return
+		if(C.wear_mask == src)
+			if(target_type != C.type)
+				C.visible_message("<span class='danger'>As [C] puts on \the [src], \his body begins to shift and contort!</span>","<span class='danger'>As you put on \the [src], your body begins to shift and contort!</span>")
+				C.transmogrify(target_type, TRUE)
+
+/obj/item/clothing/mask/morphing/attackby(obj/item/weapon/W, mob/user)
+	if(!target_type)
+		if(istype(W, /obj/item/stack/sheet/animalhide) || istype(W, /obj/item/asteroid/goliath_hide) || istype(W, /obj/item/clothing/head/bearpelt/real))
+			var/obj/item/clothing/mask/morphing/T
+			if(istype(W, /obj/item/asteroid/goliath_hide))
```

Comment:
I could use an associative list to pair skin types with the resulting masks, but associative lists only take strings as keys, right? Would I have to just compare `"[W.type]"` to the keys in the list?